doc/source/whatsnew/v3.0.0.rst

```diff
@@ -404,6 +404,7 @@ Other API changes
 - Index set operations (like union or intersection) will now ignore the dtype of
   an empty ``RangeIndex`` or empty ``Index`` with object dtype when determining
   the dtype of the resulting Index (:issue:`60797`)
+- ``np_find_common_type`` will now return ``object`` for mixed ``int64`` and ``uint64`` dtypes to avoid precision lost (:issue:`61676`, :issue:`61688`)
```

Comment:
Hi, 

I'm just skimming issues so have not looked in detail.

By glancing through I see we are changing tested behavior. This sort of suggests intentional behavior and not a bug per se. If it is a longstanding behavior we may want to deprecate first as a breaking API change?

In pandas we normally want to avoid object type, I don't see any discussion that this is an agreed way forward?